### PR TITLE
Make response consumer factory configurable

### PR DESCRIPTION
**Pull request by royaldark**
This commit updates the synchronous request method to match its async
sibling by allowing a HttpAsyncResponseConsumerFactory to be specified.

### CODE REVIEW

File: src/clj/qbits/spandex.clj [hunk 1]
```diff
@@ -192,11 +192,11 @@
           (str/index-of "application/json")))
 
 (defn ^:no-doc gzipped-entity?
-   [^HttpEntity entity]
-   (some-> entity
-           .getContentEncoding
-           .getValue
-           (str/index-of "gzip")))
+  [^HttpEntity entity]
```
Comment (royaldark) [inline src/clj/qbits/spandex.clj:195]: These lines were updated by `lein cljfmt fix`, but are otherwise unrelated to this PR.

**Comment by mpenet** [comment]
Looks good to me, thanks

**Comment by mpenet** [comment]
I modified the code slightly with later commits to avoid code duplication fyi. 

It is on clojars as 0.6.2.

**Comment by royaldark** [comment]
Awesome. Thanks for the quick turnaround!

**Pull request merged**